cmd/identity/batch.go

```diff
@@ -51,6 +44,8 @@ var (
 		MinDifficulty int    `help:"minimum difficulty to output" default:"30"`
 		Concurrency   int    `help:"worker concurrency" default:"4"`
 		OutputDir     string `help:"output directory to place keys" default:"."`
+		Permissions   uint   `default:"0700" help:"permissions to apply to output directories and/or files"`
+		Tar           bool   ` default:"false" help:"f true, identities will be output as tarballs instead of directories"`
```

Comment:
I take your point; however:

1. It's done
1. I would argue that there is value in having the option to have tarball output (without having to script it additionally)
1. The difference as far as mainenance goes keeping this:
```
func (d *deferredIdentity) writeDir(idPathPart string) error {
	idDir := filepath.Join(d.path, idPathPart)
	if err := os.MkdirAll(idDir, os.FileMode(keyCfg.Permissions)); err != nil {
		return errs.Wrap(err)
	}

	for name, data := range d.files {
		if err := ioutil.WriteFile(filepath.Join(idDir, name), data, os.FileMode(keyCfg.Permissions)); err != nil {
			return errs.Wrap(err)
		}
	}
	return nil
}
```
in sync with this:
```
func (d *deferredIdentity) writeTar(idPathPart string) error {
	tarData := new(bytes.Buffer)
	tw := tar.NewWriter(tarData)

	for name, data := range d.files {
		err := writeToTar(tw, filepath.Join(idPathPart, name), data)
		if err != nil {
			return errs.Wrap(err)
		}
	}

	if err := tw.Close(); err != nil {
		return errs.Wrap(err)
	}

	err := ioutil.WriteFile(d.path+".tar", tarData.Bytes(), os.FileMode(keyCfg.Permissions))
	if err != nil {
		return errs.Wrap(err)
	}
	return nil
}
```

Reply:
For comparison this is what it could look like without tar (ignore typos): https://gist.github.com/egonelbre/0c5662cd12c87e908761cd6364b06a59... roughly 2x less code, and clearer in what's happening.

Reply:
Creating the tar is in my opinion the smartest way when generating a bunch of identities.
Makes moving them an easier approach and it makes sure, that the necessary files stick together.
I could also think of using the tar as input for our identities in general.

Reply:
Seeing from the current ed25519 changes, it's very likely to change. Although, not much.

"it exists" is not a good argument for keeping something, especially when usability is also in question. I do understand it's not a lot of code to maintain, but I would like to ensure that people don't add features because "we can", but rather that it solves a problem.

There are few follow-up questions about `tar` like, how do we handle support on Windows, since it doesn't support it by default? The permissions for the files aren't the same as `.Save` func-s set, why the differences?

i.e. I would like to avoid these things by doing the simple thing. Create the subfolder and save as usual.

Reply:
I don't agree, but let's do only `tar` then, but also adjust the code to add default permissions that we elsewhere.

Reply:
My two cents here. I completely get the concern to leave our code base as simple as possible and eliminate extraneous features. I do think this feature is a nice user experience though. We could have another script to do this but that is more code and it's in another spot. I prefer having the code here and making it a simple flag for the user.